_posts/2020-05-12-how-to-migrate-from-capybara-webkit-to-webdrivers.md

```diff
@@ -0,0 +1,145 @@
+-    option_value = page.evaluate_script("$(\"##{field[:id]} option:contains('#{item_text}')\").val()")
+-    page.execute_script("value = ['#{option_value}']\; if ($('##{field[:id]}').val()) {$.merge(value, $('##{field[:id]}').val())}")
+-    option_value = page.evaluate_script("value")
+-    page.execute_script("$('##{field[:id]}').val(#{option_value})")
+-    page.execute_script("$('##{field[:id]}').trigger('liszt:updated').trigger('change')")
+
++    field = find_field(options[:from], visible: false)
++    find("##{field[:id]}_chzn").click
++    find("##{field[:id]}_chzn ul.chzn-results li", text: item_text).click
+  end
+```
+
+On top of updating the gem, we improved the test suite by avoiding obscure JavaScript calls.
+
+# Updating Docker Container
+
+Now that our tests are passing we can update the `Dockerfile` removing the commands that installed QT's libraries `qt5-default` and `libqt5webkit5-dev` (that will depend on the OS your are using, we have a Debian image so we just removed those packes from the `apt-get` command).
+
+Then we make sure we have Mozilla Firefox or Google Chrome installed. It was easier to install Firefox since the `firefox-esr` package was already available on the Debian Buster image we were using. You can use a Docker image which includes other browsers or install Google Chrome if you want to.
+
+So we changed the default driver to use Firefox:
+
+```ruby
+Capybara.javascript_driver = :headless_firefox
+```
+
+Now our tests can be run inside Docker and we are all green.
+
+# Updating CircleCI Config
+
+The last part of the process was to make sure the tests pass on the CI system. We only needed to make sure Firefox was available when running the test, so the easiest solution for this was to use a CircleCI's Ruby image and use the correct [variant](https://circleci.com/docs/2.0/circleci-images/#language-image-variants) so it also includes the most used browsers:
+
+```diff
+- - image: circleci/ruby:2.4.10-buster-node
++ - image: circleci/ruby:2.4.10-buster-node-browsers
+```
+
+> We did have some problems related to caching, so you may want to try to retry your CI job without caching just to make sure if you see something failing
+
+## Conclusion
+
+To sum up we got rid of an outdated dependency (`capybara-webkit`) which depended on QT and replaced it with `webdrivers` which depends on modern web browsers (e.g. Firefox). And we highly recommend you do it in your legacy Rails applications!
+
+It makes it easier for new developers to join the project, installing QT is quite complicated on some OSs (https://github.com/thoughtbot/capybara-webkit/wiki/Installing-Qt-and-compiling-capybara-webkit) and Firefox (or Chrome) are usually already installed in any web developer's environment.
+
+It also makes it easier to maintain our Rails application and it gets us closer to "the Rails way". It will help us when upgrading Rails because more recent versions of Rails uses the `webdrivers` gem by default.
```

Comment:
```suggestion
It also makes it easier to maintain our Rails application and it gets us closer to "the Rails way". It will help us when upgrading Rails because more recent versions of Rails use the `webdrivers` gem by default.
```